reference/forms/types/choice.rst

```diff
@@ -83,40 +87,98 @@ code, while the **key** is what will be shown to the user.
     and will be removed in 3.0. To read about the old API, read an older version of
     the docs.
 
+.. note::
+
+    Pre selected choices will depend on the **data** passed to the field and
+    the values of the ``choices`` option. However submitted choices will depend
+    on the **string** matching the **choice**. In the example above, the default
+    values are incrementing integers because ``null`` cannot be casted to string.
```

Comment:
because ``null`` is not a scalar value.

https://github.com/symfony/symfony/blob/2.7/src/Symfony/Component/Form/ChoiceList/ArrayChoiceList.php#L237